packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -649,8 +649,16 @@ async function handler(
           });
         }
 
+        const eventTypeHosts = eventType.hosts.length
+          ? eventType.hosts
+          : eventType.users.map((user) => ({ user, isFixed: false }));
+        const currentRRHostId = eventTypeHosts.find(
+          (host) =>
+            !host.isFixed &&
+            originalRescheduledBooking?.attendees.some((attendee) => attendee.email === host.user.email)
+        )?.user.id;
         const newLuckyUser = isSameRoundRobinHost
-          ? freeUsers.find((user) => user.id === originalRescheduledBookingUserId)
+          ? freeUsers.find((user) => user.id === (currentRRHostId ?? originalRescheduledBookingUserId))
```

Comment:
This fixes a bug related to rescheduling Round-Robin (RR) events. If a RR event had a fixed host and the "Reschedule with same Round-Robin host" option was enabled in the event settings, rescheduling a booking would exclude the original (non-fixed) host from the rescheduled booking and only the fixed host would be kept in the rescheduled booking. This fixes that issue. 


[Screencast from 22-11-24 08:25:32 PM IST.webm](https://github.com/user-attachments/assets/478614cc-e7de-44f3-9186-20586e40b007)